### ISSUE TITLE
Verification of prepared statements that haven't been executed

### ISSUE THREAD

**Issue by chbatey** [scassandra/scassandra-server#23]

**Comment by tolbertam** [comment]
@polyglotpiglet Noticed your commit and it looks pretty good!  Were you planning on opening up a PR soon?  I would be very interested in this feature for helping us test some recent changes we are making around prepared statements in the java-driver.

**Comment by tolbertam** [comment]
I gave @polyglotpiglet's approach a try and it meets my need :+1:.  Haven't formally reviewed it yet but its looking pretty good.

**Comment by chbatey** [comment]
Awesome, so all you need is the prepared statement text @tolbertam ? Not sure what else you could save apart from the time it was prepared.

**Comment by tolbertam** [comment]
Glad you brought that up, time the execution was received would be very useful since we have changes around ordering/concurrency of prepares (i.e. preparing everything async instead of 1 at a time).  I think that would be useful to have for all statements recorded in query log in general, what do you think @chbatey?   Maybe PreparedStatementPreparation can be the first one that time is included for and we can add the rest in another PR.

**Comment by polyglotpiglet** [comment]
@tolbertam, I'm pretty much done, just dealing with conflicts which I will leave until tomorrow. PR tomorrow night, if that's ok. (BST time zone :) )

**Comment by chbatey** [comment]
I agree @tolbertam  I think we should add it to them all. The only downside is if people rely on the equals method in their tests we should exclude the time field otherwise it'll be very hard for them :)

**Comment by tolbertam** [comment]
Ah, that is a good point I hadn't considered.  It isn't incredibly important for me to have timestamp right now, so maybe we can defer including the timestamp until later once we've come up with a good solution?
